### PR TITLE
chore: Adds unused lint check

### PR DESCRIPTION
**Pull request by gruyaume**
# Description

Adds the `unused` lint check and the appropriate changes to make it pass.

### CODE REVIEW

**Comment by gab-arrobo** [comment]
ok to test

File: nrfcache/nrf_cache.go [hunk 1]
```diff
@@ -7,12 +7,12 @@ package nrf_cache
 
 import (
 	"container/heap"
-	"encoding/json"
+	"sync"
+	"time"
+
```
Comment (gab-arrobo) [inline nrfcache/nrf_cache.go:12]: Why is this change needed? lint successfully completes without having to move these 2 imports

File: nrfcache/nrf_cache.go [hunk 2]
```diff
@@ -7,12 +7,12 @@ package nrf_cache
...
 	"github.com/omec-project/nrf/logger"
 	"github.com/omec-project/openapi/Nnrf_NFDiscovery"
 	"github.com/omec-project/openapi/models"
-	"sync"
-	"time"
```
Comment (gab-arrobo) [inline nrfcache/nrf_cache.go:15]: Why is this change needed? lint successfully completes without having to move these 2 imports

File: nrfcache/nrf_cache.go (hunk 1 above)
Comment (gruyaume) [reply nrfcache/nrf_cache.go:12]: This move is an outcome of running `go fmt ./...` which organizes go imports.

File: nrfcache/nrf_cache.go (hunk 2 above)
Comment (gruyaume) [reply nrfcache/nrf_cache.go:15]: See my reply above

**Review by gab-arrobo — APPROVED** [review]
+1
